### PR TITLE
Fix matter eating own head

### PR DESCRIPTION
**Pull request by flappybatpal**
[BUG]

## About the PR 
Makes eating your own head with Matter Eater function correctly instead of leaving a corpse with weird visuals and head items still attached.

## Why's this needed? 
Bug.

### CODE REVIEW

**Comment by flappybatpal** [comment]
I don't know if changing these to src is the right move but it seems like it should be?

File: code/obj/item/organs/head.dm [hunk 1]
```diff
@@ -71,16 +71,24 @@
...
-		glasses = null
-
+		src.skull = null
+		src.brain = null
+		src.left_eye = null
+		src.right_eye = null
+
+		src.head = null
+		if (donor?.head)
+			qdel(donor.head)
```
Comment (Tarmunora) [inline code/obj/item/organs/head.dm:81]: shouldn't this qdel the clothing on the head as well (e.g.: case where head-with-a-hat on the floor gets acid poured on it)

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/item/organs/head.dm (hunk 1 above)
Comment (flappybatpal) [reply code/obj/item/organs/head.dm:81]: head here is the head clothing slot because our clothing naming is insane.

**Review by flrsh — COMMENTED** [review]
I feel like the fix for this should be checking for `/obj/item/organ` in matter eater code and doing `H.organHolder.drop_organ(whatever_organ)` before qdeling it?

**Pull request merged**